_delphi_utils_python/delphi_utils/validator/dynamic.py

```diff
@@ -192,6 +192,8 @@ def validate(self, all_frames, report):
                                           "be performed"))
                     continue
 
+                self.check_reference_api_df(reference_api_df, geo_sig_df, reference_end_date)
```

Comment:
This isn't actually saving the returned padded ref API data anywhere, so subsequent checks will still use the un-padded reference data.

Reply:
Good catch! I'll add it in